packages/integrations/test/captureconsole.test.ts

```diff
@@ -0,0 +1,299 @@
+import { Event, Integration } from '@sentry/types';
+
+import { CaptureConsole } from '../src/captureconsole';
+
+const mockScope = {
+  setLevel: jest.fn(),
+  setExtra: jest.fn(),
+  addEventProcessor: jest.fn(),
+};
+
+const mockHub = {
+  withScope: jest.fn(callback => {
+    callback(mockScope);
+  }),
+  captureMessage: jest.fn(),
+  captureException: jest.fn(),
+};
+
+const getMockHubWithIntegration = (integration: Integration) => ({
+  ...mockHub,
+  getIntegration: jest.fn(() => integration),
+});
+
+// We're using this to un-monkey patch the console after each test.
+const originalConsole = Object.assign({}, global.console);
+
+describe('CaptureConsole setup', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+
+    // Un-monkey-patch the console functions
+    Object.assign(global.console, originalConsole);
+  });
+
+  it('should respect user-provided console levels', () => {
+    const captureConsoleIntegration = new CaptureConsole({ levels: ['log', 'warn'] });
+    captureConsoleIntegration.setupOnce(
+      () => undefined,
+      () => getMockHubWithIntegration(captureConsoleIntegration) as any,
```

Comment:
TBH I don't like extracting the function we intend to test into a helper. We don't gain anything from it and the tests just become more convoluted.

As for the `any` cast: I kinda struggled with that one. The `Hub` interface is huge and I didn't feel like mocking it out in its entirety when we only need a small subset. Typescript also doesn't appreciate us casting it to Hub because a bunch of methods are missing. Do you have an idea on how to do this properly?

I personally really like the [`jest-mock-extended` library](https://www.npmjs.com/package/jest-mock-extended) for type-safe mocks in unit tests. It works really well, however idk if we should introduce a dependency for this. I initially discovered this library because the [prisma orm recommends it for their unit tests](https://www.prisma.io/docs/guides/testing/unit-testing#mocking-the-prisma-client).